### PR TITLE
Out-of-order command-queue test cases for cl_khr_command_buffer

### PR DESCRIPTION
**Pull request by shajder**
Previous test case for out-of-order command-queue was moved to separate file/structure which will also handle newly added test case for simultaneous out-of-order command-queue. Task completed according to point 2.0 of issue specification:

https://github.com/KhronosGroup/OpenCL-CTS/issues/1369

### CODE REVIEW

File: test_conformance/extensions/cl_khr_command_buffer/command_buffer_out_of_order.cpp [hunk 1]
```diff
@@ -0,0 +1,357 @@
...
+
+        return CL_SUCCESS;
+    }
+
+    //--------------------------------------------------------------------------
+    clCommandQueueWrapper out_of_order_queue;
+    clCommandBufferWrapper out_of_order_command_buffer;
+
+    clEventWrapper user_event;
+    clEventWrapper wait_pass_event = nullptr;
```
Comment (EwanC) [inline test_conformance/extensions/cl_khr_command_buffer/command_buffer_out_of_order.cpp:334]: Minor - all the other wrapper variables are initialized to `nullptr` in the constructor, so `= nullptr` here looks a bit inconsistent.

**Review by EwanC — APPROVED** [review]
Codeplay's implementation doesn't support out-of-order queues, but the tests successfully skip.

File: test_conformance/extensions/cl_khr_command_buffer/command_buffer_out_of_order.cpp (hunk 1 above)
Comment (shajder) [reply test_conformance/extensions/cl_khr_command_buffer/command_buffer_out_of_order.cpp]: done

File: test_conformance/extensions/cl_khr_command_buffer/command_buffer_out_of_order.cpp [hunk 2]
```diff
@@ -33,7 +33,7 @@ struct OutOfOrderTest : public BasicCommandBufferTest
                    cl_command_queue queue)
         : BasicCommandBufferTest(device, context, queue),
           out_of_order_command_buffer(this), out_of_order_queue(nullptr),
```
Comment (shajder) [inline test_conformance/extensions/cl_khr_command_buffer/command_buffer_out_of_order.cpp:35]: @EwanC  I just noticed this one `out_of_order_command_buffer(this)` copied from former test, looks like some misprint or maybe there is some deeper sense behind ?

Comment (EwanC) [reply test_conformance/extensions/cl_khr_command_buffer/command_buffer_out_of_order.cpp:35]: Good question, the [clCommandBufferWrapper](https://github.com/KhronosGroup/OpenCL-CTS/blob/main/test_conformance/extensions/cl_khr_command_buffer/command_buffer_test_base.h#L87-L91) class takes a `CommandBufferTestBase *` in the constructor which the other wrapper classes don't. So passing `this` is intentional here.

File: test_conformance/mem_host_flags/checker_image_mem_host_write_only.hpp [hunk 3]
```diff
@@ -256,7 +256,7 @@ cl_int cImage_check_mem_host_write_only<T>::verify_RW_Image_Mapping()
     test_error(err, "clReleaseEvent error");
 
     // Map image for writing
-    T* dataPtr = (T*)clEnqueueMapImage(
```
Comment (EwanC) [inline test_conformance/mem_host_flags/checker_image_mem_host_write_only.hpp:259]: Think the formatting changes in this file are out of scope for this PR

Comment (shajder) [reply test_conformance/mem_host_flags/checker_image_mem_host_write_only.hpp:259]: indeed, `git clang-format` is culprit. Do you want me to revert it somehow ? The same thing happened with `command_buffer_profiling` branch

File: test_conformance/extensions/cl_khr_command_buffer/command_buffer_out_of_order.cpp [hunk 4]
```diff
@@ -0,0 +1,352 @@
...
+// -test case for out-of-order command-buffer
+// -test an out-of-order command-buffer with simultaneous use
+
+template <bool simultaneous_request>
+struct OutOfOrderTest : public BasicCommandBufferTest
+{
+    OutOfOrderTest(cl_device_id device, cl_context context,
+                   cl_command_queue queue)
+        : BasicCommandBufferTest(device, context, queue),
+          out_of_order_command_buffer(this), out_of_order_queue(nullptr),
```
[the remark this answers is not in the document]
Comment (shajder) [reply test_conformance/extensions/cl_khr_command_buffer/command_buffer_out_of_order.cpp]: Done

**Review by EwanC — APPROVED** [review]

**Review by aharon-abramson — APPROVED** [review]

**Review by EwanC — APPROVED** [review]
Merging as this has approvals from two implementers
